docs/development/style-guide.rst

```diff
@@ -0,0 +1,237 @@
+acceptable for longer headings designating different learning/code sections.
+
+Contractions
+============
+
+Do not use contractions in formal documentation material.
+
+Examples:
+---------
+* If you are making changes that impact ``astropy`` performance, consider adding
+  a performance benchmark.
+* You do not need to include a changelog entry.
+
+In all other materials, avoid use of contractions only when tense can be
+confused, such as in the case of “she is gone” versus “she has gone,” etc.
+
+Hyphenation
+===========
+
+Phrasal adjectives/compound modifiers placed before a noun should be hyphenated
+to avoid confusion.
+
+Examples:
+---------
+* Astronomy-related software packages.
+* Astropy provides sustainable, high-level education to the astronomy community.
+
+Hyphenated compound words should contain hyphens in plain text, but no hyphens
+in code.
+
+Example:
+--------
+* Since most of the functionality of ``astropy`` resides in sub-packages, it is
+  best to import the desired sub-package.
+
+Numbers
+=======
+
+For numbers followed by a unit or as part of a name, use the numeral.
+
+Examples:
+---------
+* 1 arcminute
+* 32 degrees
+* Gaia data release 2 catalog
+
+For all other whole numbers, follow Associated Press (AP) style: spell out
+numbers one through nine, and use numerals for 10 and higher, with numeral-word
+combinations for millions, billions, and trillions.
+
+Examples:
+---------
+* There are two ways to build Astropy documentation.
+* Follow these 11 steps.
+* Measuring astrometry for about 2 billion stars.
+
+For casual expressions, spell out the number.
+
+Example:
+--------
+* A picture is worth a thousand words.
+
+Punctuation
+===========
+
+For consistency across Astropy materials, non-U.S. punctuation will be edited
+to reflect American punctuation preferences.
+
+**Parentheses**: punctuation belonging to parenthetical material will be placed
+inside of closing parentheses, with the exception of commas to denote a small
+pause coming after parenthetical material, and periods when parenthetical
+material is included within another sentence.
+
+Examples:
+---------
+* (For full contributor guidelines, see our documentation.)
+* Once you open a pull request (which should be opened against the ``master``
+  branch), please make sure to include the following.
+* In some cases, most of the required functionality is contained in a single
+  class (or a few classes).
+
+**Quotation marks**: periods and commas will be placed inside of closing
+quotation marks, whether double or single.
+
+Examples:
+---------
+* Chief among these terms is the concept of a “coordinate system.”
+* Because of the likelihood of confusion between these meanings of “coordinate
+  system,” `~astropy.coordinates` avoids this term wherever possible.
+
+**Hyphens vs. En Dashes vs. Em Dashes**
+
+Hyphens (-) should be used for phrasal adjectives and compound words (see
+Hyphenation above).
```

Comment:
You can use RST-style linking here if you wish. See first answer in https://stackoverflow.com/questions/48888604/how-to-create-a-link-to-a-header-in-restructuredtext .